src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java

```diff
@@ -0,0 +1,66 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2015 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+package com.puppycrawl.tools.checkstyle.checks.header;
+
+import java.util.regex.PatternSyntaxException;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ *
+ * @author richter
+ */
+public class RegexpHeaderCheckTest
+{
+
+    public RegexpHeaderCheckTest()
+    {
+    }
+
+    /**
+     * Test of setHeader method, of class RegexpHeaderCheck.
+     */
+    @Test
+    public void testSetHeader()
+    {
+        // check null passes
+        RegexpHeaderCheck instance = new RegexpHeaderCheck(); // recreate for
+            // each test because multiple invokations fail
+        String header = null;
+        instance.setHeader(header);
+        // check empty string passes
+        instance = new RegexpHeaderCheck();
+        header = "";
+        instance.setHeader(header);
+        // check valid header passes
+        instance = new RegexpHeaderCheck();
+        header = "abc.*";
+        instance.setHeader(header);
+        // check invalid header passes
+        instance = new RegexpHeaderCheck();
+        header = "^/**\\n * Licensed to the Apache Software Foundation (ASF)";
+        try {
+            instance.setHeader(header);
+            Assert.fail(String.format("%s should have been thrown", PatternSyntaxException.class));
```

Comment:
please , do not follow that advice, "expect" parameter usage is forbidden as it make a mess in logs.